src/freshrssurlreader.cpp

```diff
@@ -0,0 +1,86 @@
+#include "freshrssurlreader.h"
+
+#include "configcontainer.h"
+#include "fileurlreader.h"
+#include "logger.h"
+#include "remoteapi.h"
+#include "utils.h"
+
+namespace newsboat {
+
+FreshRssUrlReader::FreshRssUrlReader(ConfigContainer* c,
+	const std::string& url_file,
+	RemoteApi* a)
+	: cfg(c)
+	, file(url_file)
+	, api(a)
+{
+}
+
+FreshRssUrlReader::~FreshRssUrlReader() {}
+
+#define STARRED_ITEMS_URL \
+	"/reader/api/0/stream/contents/user/-/state/com.google/starred"
+
+#define ADD_URL(url, caption)				  \
+	do {								  \
+		tmptags.clear();			  \
+		urls.push_back((url));		  \
+		tmptags.push_back((caption)); \
+		tags[(url)] = tmptags;		  \
+	} while (0)
```

Comment:
Suggestion: since these macros are only used in a single place, I'd inline them, making the code simpler.